cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/SourceControlManagementService.java

```diff
@@ -57,8 +59,14 @@ private NamespaceTable getNamespaceTable(StructuredTableContext context) throws
     return new NamespaceTable(context);
   }
 
-  public void setRepository(NamespaceId namespace, RepositoryConfig repository) throws NamespaceNotFoundException {
+  public void setRepository(NamespaceId namespace, RepositoryConfig repository)
+    throws NamespaceNotFoundException, InvalidRepositoryConfigException {
     accessEnforcer.enforce(namespace, authenticationContext.getPrincipal(), StandardPermission.UPDATE);
+    if (repository == null) {
```

Comment:
by convention, we always assume an argument is not null unless it is annotated with `@Nullable`. This null check is more like something that should be checked in the handler, resulting in a 400

Reply:
moved to the handler